docarray/array/mixins/io/pushpull.py

```diff
@@ -1,24 +1,184 @@
+import json
 import os
+import os.path
 import warnings
+from collections import Counter
 from pathlib import Path
-from typing import Dict, Type, TYPE_CHECKING, Optional
+from typing import Dict, Type, TYPE_CHECKING, List, Optional, Any
+
+import hubble
+from hubble import Client
+from hubble.client.endpoints import EndpointsV2
+
 
 from docarray.helper import get_request_header, __cache_path__
 
 if TYPE_CHECKING:
     from docarray.typing import T
 
 
+def _get_length_from_summary(summary: List[Dict]) -> Optional[int]:
+    """Get the length from summary."""
+    for item in summary:
+        if 'Length' == item['name']:
+            return item['value']
+
+
 class PushPullMixin:
     """Transmitting :class:`DocumentArray` via Jina Cloud Service"""
 
     _max_bytes = 4 * 1024 * 1024 * 1024
 
+    @classmethod
+    @hubble.login_required
+    def cloud_list(cls, show_table: bool = False) -> List[str]:
+        """List all available arrays in the cloud.
+
+        :param show_table: if true, show the table of the arrays.
+        :returns: List of available DocumentArray's names.
+        """
+        from rich import print
+
+        result = []
+        from rich.table import Table
+        from rich import box
+
+        resp = Client(jsonify=True).list_artifacts(
+            filter={'type': 'documentArray'}, sort={'createdAt': 1}
+        )
+
+        table = Table(
+            title=f'You have {resp["meta"]["total"]} DocumentArray on the cloud',
+            box=box.SIMPLE,
+            highlight=True,
+        )
+        table.add_column('Name')
+        table.add_column('Length')
+        table.add_column('Access')
+        table.add_column('Created at', justify='center')
+        table.add_column('Updated at', justify='center')
+
+        for da in resp['data']:
+            if da['type'] == 'documentArray':
```

Comment:
should be enough to do `filter={'type': 'documentArray'},`, otherwise, maybe it's better not to hide the bug

Reply:
yes, it was a legacy bug from hubble-sdk, now it solved i will refactor this one